sqlmesh/core/console.py

```diff
@@ -1024,14 +1026,24 @@ def show_schema_diff(self, schema_diff: SchemaDiff) -> None:
 
         self.console.print(tree)
 
-    def show_row_diff(self, row_diff: RowDiff, show_sample: bool = True) -> None:
+    def show_row_diff(
+        self, row_diff: RowDiff, show_sample: bool = True, check_grain: bool = False
+    ) -> None:
         source_name = row_diff.source
         if row_diff.source_alias:
             source_name = row_diff.source_alias.upper()
         target_name = row_diff.target
         if row_diff.target_alias:
             target_name = row_diff.target_alias.upper()
 
+        if check_grain and (
+            row_diff.stats["null_grain_count"] > 0
+            or row_diff.stats["join_count"] != row_diff.stats["distinct_grain_count"]
+        ):
+            self.console.print(
+                "[b][red]Grain should have unique and not-null audits for accurate results.[/red][/b]"
```

Comment:
the null grains is cheap, there's no reason why that has to be covered by the flag

Reply:
Right, I will have this check then always on by default